vm/src/builtins/type.rs

```diff
@@ -405,6 +405,35 @@ impl PyType {
             .unwrap_or_else(|| vm.ctx.new_str(self.name().deref()).into())
     }
 
+    #[pyproperty(magic, setter)]
+    fn set_qualname(&self, value: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        if value.is(&vm.ctx.none()) {
+            return Err(vm.new_type_error(format!(
+                "cannot delete '{}' attribute of immutable type '{}'",
+                "__qualname__",
+                self.name()
+            )));
+        };
+        if !self.slots.flags.has_feature(PyTypeFlags::HEAPTYPE) {
+            return Err(vm.new_type_error(format!(
+                "cannot set '{}' attribute of immutable type '{}'",
+                "__qualname__",
+                self.name()
+            )));
+        };
```

Comment:
Maybe it is trivial, It seems this `HEAPTYPE` flag check should precede a `None` check according to cpython's work.
```python
# cpython
>>> del int.__qualname__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot set '__qualname__' attribute of immutable type 'int'
# rustpython
>>>>> del int.__qualname__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot delete '__qualname__' attribute of immutable type 'int'
```

Reply:
Oh, I missed it. I will check `HEAPTYPE` before `None`. Thank you